### PR TITLE
feat: added assetchain testnet

### PR DESCRIPTION
**Pull request by CeoFred**
## PR-Codex overview
The focus of this PR is to add the `AssetChain Testnet` to the project's supported chains.

### Detailed summary
- Added `AssetChain Testnet` to supported chains
- Created `assetChainTestnet` definition in `src/chains/definitions/assetChainTestnet.ts`
- Defined chain properties like `id`, `name`, `nativeCurrency`, `rpcUrls`, `blockExplorers`, `testnet`, and `contracts`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: .vscode/settings.json
Comment (jxom) [inline .vscode/settings.json:1]: Revert this change.

File: pnpm-lock.yaml
Comment (jxom) [inline pnpm-lock.yaml:1]: Revert this change.

Comment (CeoFred) [reply pnpm-lock.yaml:1]: I've deleted the file, i guess locks files should be added to gitignore. @jxom

**Comment by KodeSage** [comment]
@jxom please review this pr again.

File: pnpm-lock.yaml
Comment (jxom) [inline pnpm-lock.yaml:1]: Please revert this change.

**Comment by jxom** [comment]
We need the lockfile.

**Comment by KodeSage** [comment]
so it's good now to merge or is the lockfile still needed @jxom ?

**Comment by jxom** [comment]
We need the lockfile. Please reopen when done.

**Pull request closed without merging**